tutorial/task_presentation.md

```diff
@@ -0,0 +1,131 @@
+# task-parallelism with Pyccel
+
+## A task
+
+Parallel program is viewed as a collection of tasks that communicate by sending
+messages to each other through dependencies.
+A task consists of an executable unit or a unit of computation (think of it as a program),
+together with its local memory and a collection of I/O ports. that can/should execute in
+parallel with other tasks.
+
+## Context
+
+A problem can be borken into multiple parts each part can be represented as a task, each
+task is a separate unit of work that can be independant or may take some dependencies from other
+tasks. task with dependencies may only start when all antecedents have completed. 
+So in general the dependencies between tasks can be represented as a directed acyclic graph
+where tasks form the vertices edge are the dependencies between task.
+
+## Usage:
+Using the decorator task allows pyccel to recognize that the current function will
+be used in task parallelism and automatically calculate its dependencies.
+
+```Python
+@task
+def A():
+    # do something
+```
+
+A task function can only run in parallel if the parent function is a task.
+```Python
+@task('master')
+def main():
+    a = A()
+```
+
+```Python
+@task('master')
+def recursive() -> int:
+    a = recursive()
+    # do something
+```
+
+
+A block of tasks that need to run in parallel is by the first occurrence of a task 
+function in (FunctionCall or Assign) until the first non task block of the end of the program.
+So avoid writing a codeblock that depend on a task result between tasks calls.
```

Comment:
Maybe clearer:
```
The parallel block generated by pyccel will only contain function calls and assignments which save the result of a function call. As a result it is advisable to avoid writing a code block which depends on a task result between task calls; otherwise the parallel block will be split into multiple blocks
```